### DISCUSSION TITLE
How to reproduce RTP stream?

### DISCUSSION THREAD

**Discussion by joaomfariamartins** [bluenviron/mediamtx#311] [Questions]
## Which version are you using?

v0.13.3

## Question

I have a DVR that has a camera and it uses a SIM card to communicate with the external world. If the DVR receives an instruction to perform live streaming, the DVR starts sending RTP packets through UDP or TCP to a server.

How can I view the RTP stream? I think I don't understand how RTP and RTSP are connected. I understand the theory but I find it difficult to put it into practice. 
```
RTSP is a realtime streaming protocol. Meaning, you can stream whatever you want in real-time. So you can use it to stream LIVE content. 

RTP is a transport protocol which is used to transport media data which is negotiated over RTSP. 

You use RTSP to control media transmission over RTP. You use it to setup, play, pause, teardown the stream.

Source: https://stackoverflow.com/a/4303565/8284083
```

I'm running the RTSP with Docker (`docker run --rm -it --network=host aler9/rtsp-simple-server`) - terminal output:
```
docker run --rm -it --network=host aler9/rtsp-simple-server
2021/02/15 12:00:23 [I] [0/0/0] rtsp-simple-server v0.13.3
2021/02/15 12:00:23 [I] [0/0/0] [UDP/RTP listener] opened on :8000
2021/02/15 12:00:23 [I] [0/0/0] [UDP/RTCP listener] opened on :8001
2021/02/15 12:00:23 [I] [0/0/0] [TCP/RTSP listener] opened on :8554
```

My DVR is sending RTP packets using UDP to the `{my_IP}:8000`. How can I stream the video using for example VLC media player?

I'm a newbie in this field and I might be confusing some concepts. Any help or advice is welcome.

**Comment by elcapitan26** [comment]
RTP playback requires an sdp file that describes the stream. It can then be opened in VLC. The instructions here are relevant: https://www.matrox.com/en/file/9551/download?token=GMy71qbs

**Comment by aler9** [comment]
RTSP uses RTP internally, but unfortunately pure RTP streams are not RTSP streams, and must be converted into RTSP streams to be accepted and served by rtsp-simple-server.

The first step consists in obtaining the SDP file corresponding to the RTP stream - the SDP file contains infos about

* codec
* ip
* port

of the stream.

After you have it, try to open it with VLC and see if it works.

Then, you can read the RTP stream with ffmpeg and convert it into RTSP:

```
ffmpeg -i myfile.sdp -c copy -f rtsp rtsp://serverip:8554/serverpath
```

[a comment by an automated account is not shown]
